site-src/geps/gep-851.md

```diff
@@ -0,0 +1,80 @@
+# GEP-851: Allow Multiple Certificate Refs per Gateway Listener
+
+* Issue: [#851](https://github.com/kubernetes-sigs/gateway-api/issues/851)
+* Status: Implementable
+
+## TLDR
+
+Replace `CertificateRef` field with a `CertificateRefs` field in Gateway
+Listeners.
+
+## Goals
+
+* Enable specifying RSA and ECDSA certs on the same Listener.
+* Provide a path to future expansion that could include:
+  * Support for referencing certificates for different hostnames (maybe as part
+    of a self-service TLS approach).
+  * Support for including new and old certificates as part of renewal process.
+
+## Non-Goals
+
+* Define how future expansion could work.
+
+## Introduction
+
+As described above, there are a number of potential use cases for attaching
+multiple certificates to a Gateway Listener. The most straightforward reason for
+that involves attaching RSA and ECDSA certs. Although this is not a very common
+use case, it is clearly understood and broadly supported.
+
+This change will enable future expansion of the API. It is very likely that we
+will want to expand this API to support at least one of the additional
+capabilities described above. Starting with a list of certificate refs instead
+of a single certificate ref will make future expansion much simpler.
+
+## API
+
+The `CertificateRef` field in `GatewayTLSConfig` would be replaced with the
+following `CertificateRefs` field:
+
+```go
+    // CertificateRefs contains a series of references to Kubernetes objects that
+    // contains TLS certificates and private keys. These certificates are used to
+    // establish a TLS handshake for requests that match the hostname of the
+    // associated listener.
+    //
+    // A maximum of 2 certificates may be referenced from this field - one RSA
```

Comment:
I don't think we should be this prescriptive. This is going to cause problems when moving to having more than one certificate of differing types etc.

I think we should just leave it not specified that this requires a certain kind of cert etc.

Reply:
👍 removed.